### PR TITLE
Add istio/istio postsubmit testgrid and enable email alert

### PR DESCRIPTION
**Pull request by yutongz**

### CODE REVIEW

File: testgrid/config/config.yaml [hunk 1]
```diff
@@ -1903,8 +1903,17 @@ test_groups:
...
-  gcs_prefix: istio-prow/api-postsubmit
+- name: istio-istio-postsubmit
+  gcs_prefix: istio-prow/istio-postsubmit
+  num_failures_to_alert: 1
+- name: istio-e2e-suite-rbac-no_auth
+  gcs_prefix: istio-prow/e2e-suite-rbac-no_auth
+  num_failures_to_alert: 3
+- name: istio-e2e-suite-rbac-auth
+  gcs_prefix: istio-prow/e2e-suite-rbac-auth
+  num_failures_to_alert: 3
```
Comment (BenTheElder) [inline testgrid/config/config.yaml]: you may also want to consider alerting on stale tests

**Comment by BenTheElder** [comment]
@yutongz too late, but you may want to tweak these in the future, since sen is OOO atm just `/assign @bentheelder` on any follow ups and I'll be sure to review :-)

**Comment by yutongz** [comment]
@BenTheElder Yeah, you are right, I will add them on later :)

I noticed that in testgrid https://k8s-testgrid.appspot.com/istio, it seems it doesn't pick up our tests. While we do have tests actively running https://k8s-testgrid.appspot.com/istio.

It might because we restarted the prow cluster (which we shouldn't), and then job count reset. Thought?

**Comment by BenTheElder** [comment]
@yutongz yes that's likely why, you can fix the build numbers if they're on prow by updating tot, from the prow cluster (I use a curl pod) you can run `curl -X POST -d "$NEXT_BUILD_NUMBER" http://tot/vend/$JOB_ID` eg: `curl -X POST -d "5413" http://tot/vend/pull-kubernetes-cross` xref: https://github.com/kubernetes/test-infra/issues/2996#issuecomment-345488656

**Comment by BenTheElder** [comment]
@yutongz I did some more digging into this and it looks like you had significantly higher build numbers than you are at now (looks like before well into `1xxx` while currently at `1xx`), I believe testgrid has some caching mechanisms and doesn't expect build numbers to be reused.

For example I was able to find (with `gsutil ls gs://istio-prow/...`):
http://gcsweb.k8s.io/gcs/istio-prow/e2e-suite-rbac-no_auth/1795/
http://gcsweb.k8s.io/gcs/istio-prow/e2e-suite-rbac-auth/1779/
http://gcsweb.k8s.io/gcs/istio-prow/e2e-cluster_wide-auth/891/
http://gcsweb.k8s.io/gcs/istio-prow/istio-postsubmit/179/

The most reasonable fix is probably just finding the last build number used for each job and then bumping `tot` so that prow will give new jobs fresh build IDs in the correct sequence (see above). 

Though the bucket contents are quite strange, there are large gaps in the build IDs. Have you removed results from the bucket or... ? How exactly does `e2e-suite-rbac-auth` jump from 178 to 1726?

**Comment by yutongz** [comment]
Hmm, IC. What does this command do? I am wondering why it's not picking up the build number from latest-build.txt?

Anyway, I manager to make it work by delete the log files from last counting cycle. The wired thing is `istio-postsubmit` found whole bunch of job runs we have but E2E-tests only tracked few. 
https://k8s-testgrid.appspot.com/istio#Summary

log folders in gcs are clear and only includes numbers in this cycle, from 1 to 179/180 or so, matched the latest runs we have now.

**Comment by BenTheElder** [comment]
@yutong `tot` is the build ID vending component, posting to that URL sets the ID, GET vends a new one.
latest-build.txt are specific to Gubernator and testgrid, prow doesn't currently know or care about anything you do in GCS, that is all handled by bootstrap at the moment.

Again, I suspect testgrid does some caching, so restarting your build IDs may cause it to not fetch new data for existing IDs.

**Comment by yutongz** [comment]
https://k8s-testgrid.appspot.com/istio#e2e-cluster_wide-auth
This one failed more than 5 times in a row, curious the reason why we don't get the email notification =)

**Comment by BenTheElder** [comment]
@yutongz I'm not sure why, will ask Michelle tomorrow. I've successfully gotten alerts before 😕 

@krzyzacy and I sacrificed our personal emails against highly flaky jobs for a bit when external email alerts was first built and since then we've started enabling them for jobs we need to monitor bit by bit. Not sure why this didn't work 😕

**Comment by yutongz** [comment]
https://github.com/kubernetes/test-infra/pull/5908

And what email account testgrid is using, maybe you can take a look at sent history to find some clues?

**Comment by BenTheElder** [comment]
@yutongz testgrid uses `noreply+testgrid@google.com` and I did not get any alerts after joining your group, I specifically filter and label these:
screen shot 2017-12-12 at 9 45 37 am
(highlight is because I searched for "testgrid" to dig one out)

**Comment by yutongz** [comment]
I didn't get any email from address either.

**Comment by michelle192837** [comment]
So in order to avoid spamming alerts in a few particular (but annoying) edge cases, we don't alert when a test has been failing since the beginning of the dashboard. Since that tab has some passing results now, any further failures should be correctly alerted for when they fail enough times in a row (but let me know if that's not the case!)
